crates/net/ress/src/connection.rs

```diff
@@ -296,6 +311,13 @@ where
 
             return Poll::Pending;
         }
+
+        // Terminating the connection.
+        this.terminated = true;
+        let _ = this
+            .active_connections
+            .fetch_update(Ordering::Relaxed, Ordering::Relaxed, |c| Some(c.saturating_sub(1)));
```

Comment:
this is a bit flawed, because now the counter only resets if the connection is being terminated internally.

you can wrap this in a helper type that does this on drop so that even if the connection is terminated externally the counter resets